pkg/cloudprovider/aws/aws.go

```diff
@@ -99,6 +108,44 @@ func (c *CloudProvider) Refresh() error {
 	return c.RegisterNodeGroups(ids...)
 }
 
+type Instance struct {
+	id          string
+	ec2Instance *ec2.Instance
+}
+
+func (c *CloudProvider) GetInstance(node *v1.Node) (cloudprovider.Instance, error) {
+	var instance Instance
+
+	id := providerIdToInstanceId(node)
+
+	instanceIds := make([]*string, 0, 1)
+	instanceIds = append(instanceIds, &id)
+	input := &ec2.DescribeInstancesInput{
+		InstanceIds: instanceIds,
+	}
+
+	result, err := c.ec2_service.DescribeInstances(input)
+
+	if err != nil {
```

Comment:
This logic is a little weird, you should just return the error early if there is a problem:
```go
result, err := c.ec2_service.DescribeInstances(input)
if err != nil {
	return instance, err
}
```